tests/stress_tests/memcheck_tests/tests.cpp

```diff
@@ -106,3 +106,60 @@ INSTANTIATE_TEST_CASE_P(MemCheckTests, MemCheckTestSuite,
                         ::testing::ValuesIn(
                                 generateTestsParams({"devices", "models"})),
                         getTestCaseName);
+
+TEST_P(MemCheckTestSuite, inference_with_streams) {
+    const auto nstreams = 2;
+    log_info("Inference of InferRequest from network: \"" << model
+                                                          << "\" with precision: \"" << precision
+                                                          << "\" for device: \"" << device << "\""
+                                                          << "\" with streams: \"" << nstreams);
+    if ((device != "CPU") && (device != "GPU"))
+        throw std::invalid_argument("This device is not supported");
+
+    auto test_pipeline = [&] {
+        MemCheckPipeline memCheckPipeline;
+
+        std::map<std::string, std::string> config;
+        const std::string key = device + "_THROUGHPUT_STREAMS";
+        config[device + "_THROUGHPUT_STREAMS"] = std::to_string(nstreams);
+
+        Core ie;
+        ie.GetVersions(device);
+        ie.SetConfig(config, device);
+
+        InferRequest inferRequest;
+
+        CNNNetwork cnnNetwork = ie.ReadNetwork(model);
+        ExecutableNetwork exeNetwork = ie.LoadNetwork(cnnNetwork, device);
+        auto batchSize = cnnNetwork.getBatchSize();
+        batchSize = batchSize != 0 ? batchSize : 1;
+        const ConstInputsDataMap inputsInfo(exeNetwork.GetInputsInfo());
+
+        unsigned int nireq = nstreams;
+        try {
+            nireq = exeNetwork.GetMetric(METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS)).as<unsigned int>();
+        } catch (const std::exception &ex) {
+            log_err("Failed to query OPTIMAL_NUMBER_OF_INFER_REQUESTS");
+        }
+        for (int counter = 0; counter < nireq; counter++) {
+            inferRequest = exeNetwork.CreateInferRequest();
+            fillBlobs(inferRequest, inputsInfo, batchSize);
+
+            inferRequest.Infer();
```

Comment:
Sstreams imply async execution, so just loop and run'm all with StartAsync, and count arrival (completion) in the callback (or, simpler, wait them in another loop)